client/src/app/api/models.ts

```diff
@@ -118,7 +118,7 @@ export interface Application {
   identities?: Ref[];
   repository?: Repository;
   binary?: string;
-  migrationWave?: Ref;
+  migrationWave: Ref | null;
```

Comment:
Why the change to null here?

Reply:
Because migrationWave is a field that exists in the record meanwhile its value can be `null`.

Reply:
Undefined typically means a variable has been declared but not defined yet. This aligns with how the hub uses our payloads that we pass. We don't pass the hub "null" values for fields that do not exist. We simply omit the field when making the request. Adding explicit null values is useful when we need a special user defined null case that sits outside of our normal usage.

Reply:
Take a look at the payload we send to the hub 

![image](https://github.com/konveyor/tackle2-ui/assets/11218376/765c3e48-72a7-4423-b815-0538b9b9cf48)

Reply:
```suggestion

export interface Application {
  id: number;
  name: string;
  description?: string;
  comments?: string;
  businessService?: Ref;
  tags?: TagRef[];
  owner?: Ref;
  contributors?: Ref[];
  review?: Ref;
  identities?: Ref[];
  repository?: Repository;
  binary?: string;
  migrationWave?: Ref;
}
```

Reply:
How the hub stores the values in the database doesn't really matter to us. We are working in TS world where the creators/maintaners  themselves have recommended  usage "undefined" and not use "null". I personally find that changing our pattern to "null" at this stage doesn't make much sense as both are acceptable and can be valid. There is no real strong point one way or another but I enjoy using the '?' syntax as I have come to find it makes the code more readable.

Reply:
Here is the reason optional fields are only convenient at creation time:

If an application is updated (i.e AppicationForm) and the `migrationWave` field is omitted then the relationship is silently severed and the application disappears from the migration wave:

```json
{
	"binary": "::",
	"comments": "",
	"contributors": [],
	"description": "App1",
	"id": 1,
	"migrationWave": {
		"id": 3,
		"name": "Wave1"
	},
	"name": "App1",
	"tags": []
}
```

Reply:
So maybe we're in a special case where `migrationWave` is a mandatory field.
The rest might be a question of view but I'll tend to make any recorded field not optional and avoid testing all the time if such field is defined and focus on the job to know if what value it contains or not.

Reply:
I  think I see your point. So instead of having to add 

```
       migrationWave: application?.migrationWave,
````
to the update payload when updating an application from the form that has no migrationWave field included ( it should already exist on the application from other views. just like we have other fields not in the application edit form (review, bucket, etc)), we can do 

````
    if (application) {
      updateApplication({ ...application, ...payload });
    } else {
      createApplication(payload);
    }
````
to make sure we are not stripping any data from the application object by not including it in the update request.

Reply:
@gildub Yes that is what they decided on for their project. 


https://github.com/Microsoft/TypeScript/wiki/Coding-guidelines#null-and-undefined